frontend/app/src/components/nft/NftGalleryItem.vue

```diff
@@ -11,41 +13,147 @@ const props = defineProps({
+
+const frontendStore = useFrontendSettingsStore();
+
+const { whitelistedDomainsForNftImages } = storeToRefs(frontendStore);
+const { updateSetting } = frontendStore;
+
+const imageUrlSource: ComputedRef<string | null> = computed(
+  () => get(item).imageUrl
+);
+
 const name = computed(() =>
   get(item).name ? get(item).name : get(item).collection.name
 );
 
-const imageUrl = computed(
-  () => get(item).imageUrl ?? './assets/images/placeholder.svg'
+const renderImage: ComputedRef<boolean> = computed(() => {
+  const image = get(imageUrlSource);
+
+  if (!image) {
+    return true;
+  }
+
+  return shouldRenderImage(image);
+});
+
+const imageUrl = computed(() => {
+  const image = get(imageUrlSource);
+
+  if (!image || !get(renderImage)) {
+    return './assets/images/placeholder.svg';
+  }
+
+  return image;
+});
+
+const isMediaVideo = computed(() => isVideo(get(imageUrl)));
+
+const { tc } = useI18n();
+const css = useCssModule();
+
+const domain: ComputedRef<string | null> = computed(() =>
+  getDomain(get(imageUrlSource) || '')
 );
 
-const isMediaVideo = computed(() => isVideo(get(item).imageUrl));
+const { show } = useConfirmStore();
+
+const showAllowDomainConfirmation = () => {
+  show(
+    {
+      title: tc(
+        'general_settings.nft_setting.update_whitelist_confirmation.title'
+      ),
+      message: tc(
+        'general_settings.nft_setting.update_whitelist_confirmation.message',
+        2,
+        {
+          domain: get(domain)
+        }
+      )
+    },
+    allowDomain
+  );
+};
+
+const allowDomain = () => {
+  const domainVal = get(domain);
+
+  if (!domainVal) {
+    return;
+  }
+
+  const newWhitelisted = [
+    ...get(whitelistedDomainsForNftImages),
+    domainVal
+  ].filter(uniqueStrings);
+
+  updateSetting({ whitelistedDomainsForNftImages: newWhitelisted });
+};
 </script>
 
 <template>
   <v-card class="mx-auto overflow-hidden">
-    <base-external-link :href="item.externalLink">
-      <video
-        v-if="isMediaVideo"
-        controls
-        width="auto"
-        :src="imageUrl"
-        :style="{
-          'background-color': `#${item.backgroundColor}`
-        }"
-      />
-      <v-img
-        v-else
-        :src="imageUrl"
-        contain
-        aspect-ratio="1"
-        :style="{
-          'background-color': `#${item.backgroundColor}`
-        }"
-      />
-    </base-external-link>
+    <div :class="css.wrapper">
+      <v-tooltip top :disabled="renderImage" max-width="200" open-delay="200">
+        <template #activator="{ on }">
+          <div v-on="on">
+            <base-external-link :href="item.externalLink">
+              <video
+                v-if="isMediaVideo"
+                controls
+                width="auto"
+                :src="imageUrl"
+                :style="
+                  renderImage
+                    ? {
+                        'background-color': `#${item.backgroundColor}`
+                      }
+                    : {}
```

Comment:
this style is the same as the one bellow right? any chance we can move it to a computed prop?